### PR TITLE
Fix EntityPicker allowing clear value when required #16993

### PR DESCRIPTION
**Pull request by morales-martin**
## Hey, I just made a Pull Request!

Fixed a bug (#16993) in the OwnedEntityPicker\/EntityPicker component that allowed for empty values when field is set as required. This bug occurs only after a user fills the OwnedEntityPicker\/EntityPicker field, clears it, and then continues to the next form step.

This bug was addressed by passing a value of undefined instead of an empty string to the onChange event when the field is cleared. Using undefined instead of an empty string triggers validation errors when the field is required and empty.

Existing tests for the EntityPicker component were used to verify new functionality and to make sure the new logic did not interfere with free solo or non-required EntityPicker components. 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Parsifal-M — COMMENTED** [review]
I'm not entirely familiar with this, but do we already test this in the test cases? Or would we need to update the tests to account for `undefined`?

**Comment by morales-martin** [comment]
> I'm not entirely familiar with this, but do we already test this in the test cases? Or would we need to update the tests to account for `undefined`?

Hi - I looked over the current tests for EntityPicker and it didn't seem like there were existing tests for entering blank/clear input values. I just added tests to my commit for required and non-required solo/non-solo instances of  EntityPicker. 

With the new logic using `undefined `instead of `''`, the actual value of the dropdown component will still resolve to `''`. What is changing is the value that is being passed to the onChange function (`undefined`). This triggers the `@rjsf` errorSchema to recognize the `undefined `value and raise a required validation error. 

In other words, even if undefined is being passed to the onChange function, the component value will still resolve to `''`. And with this commit, validation for required EntityPickers will be correctly identified for blank inputs.

**Review by Parsifal-M — APPROVED** [review]
That makes sense, thanks for the detailed response! Hopefully @Rugvip feels the same way 😅

**Comment by morales-martin** [comment]
Hi @Rugvip! Any chance you can review this PR? I can't assign it to you

**Comment by morales-martin** [comment]
Unstale.
